### PR TITLE
Add travis bionic build

### PR DESCRIPTION
**Pull request by ForeverRainbow**
This adds a bionic build to travis. 

In order to get it to build on bionic I had to install gcc9 from the "testing" toolchain repo. I have also removed the cmake build from source since it is no longer needed.

~~Build times are _much_ faster now, testing this it took 8-9 minutes to build the linux ones vs the 20+ minutes it used to take~~ nvm I am wrong, this does not seem to affect build times

### CODE REVIEW

**Comment by giuspen** [comment]
The Linux was already about 8 minutes, the windows build about twice the Linux while the bottleneck is tha Mac os that takes half an hour

File: .travis.yml [hunk 1]
```diff
@@ -59,12 +70,7 @@ install:
       linux)
```
Comment (giuspen) [inline .travis.yml:70]: I think the whole linux) case could go here

File: .travis.yml [hunk 2]
```diff
@@ -23,6 +29,11 @@ before_install:
         sudo apt install build-essential libgtkmm-3.0-dev xvfb
         sudo apt install libgtksourceviewmm-3.0-dev libxml++2.6-dev libsqlite3-dev
         sudo apt install libcpputest-dev gettext python3-lxml libgspell-1-dev libcurl4-openssl-dev
+        
+        sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
+        sudo apt update
+        sudo apt install gcc-9 g++-9 -y
+        sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 60 --slave /usr/bin/g++ g++ /usr/bin/g++-9
```
Comment (giuspen) [inline .travis.yml:36]: Why is this necessary? Is this only for Travis or also for any bionic Ubuntu build?

Comment (ForeverRainbow) [reply .travis.yml:36]: Any bionic build. The problem is that the gcc which ships with ubuntu 18 does not support c++17, I havn't used it in a while but I would guess its gcc6.

Comment (ForeverRainbow) [reply .travis.yml:36]: I didn't mention it before since 1. I forgot and 2. cherrytree needs a C++17 compiler anyway

**Comment by ForeverRainbow** [comment]
@giuspen Ah my mistake then, guess `brew` installing all those packages really slows mac down

**Comment by giuspen** [comment]
I experimented something in the Mac os to try and cache the brew build but it hasn't worked, maybe some trick will come out in future

**Comment by giuspen** [comment]
Thanks!
